codec/type_parser.go

```diff
@@ -33,31 +29,17 @@ func NewTypeParser[T any, X any, Y bool]() *TypeParser[T, X, Y] {
 // Register registers a new type into TypeParser [p]. Registers the type by using
 // the string representation of [o], and sets the decoder of that index to [f].
 // Returns an error if [o] has already been registered or the TypeParser is full.
-func (p *TypeParser[T, X, Y]) Register(o T, f func(*Packer, X) (T, error), y Y) error {
-	if p.index == consts.MaxUint8 {
+func (p *TypeParser[T, X, Y]) Register(id uint8, f func(*Packer, X) (T, error), y Y) error {
+	if uint8(len(p.indexToDecoder)) == consts.MaxUint8 {
```

Comment:
nit: Do we want to enforce items are added monotonically?

I guess we don't need to...so we probably shouldn't 🤔

Reply:
Hum, if you are pointing the `if` statement. It is not related to a monotically insert.
We just check if the number of stored ID is equal to the maxUint8. When equal, it means we already stored all the possible value represented by a uint8.
By the way it should be `.... == consts.MaxUint8 + 1`

Reply:
Yep, I left this comment here just to note the change (which used to be enforced by the TypeParser).

It is ok to ignore.

Reply:
Nice catch on the +1.